pipeline/test/test_processor.py

```diff
@@ -0,0 +1,788 @@
+import os
+from pathlib import Path
+
+import pandas as pd
+
+from unittest.mock import patch
+from extract.document import Document, Page, TextBlock
+from processor.postprocessor import (
+    HyphenationPostProcessor,
+    AdobeListGroupingPostProcessor,
+    AdobeTextStylingPostProcessor,
+)
+import pytest
+
+
+@pytest.fixture
+def test_input_path() -> Path:
+    return (
+        Path(__file__).parent / "data/cclw-1318-d7f66920a18e4ddf94c83cf21fa2bcfa.json"
+    )
+
+
+# Keep these as fixtures in case they get more complex in the future.
+@pytest.fixture
+def adobe_list_postprocessor() -> AdobeListGroupingPostProcessor:
+    return AdobeListGroupingPostProcessor()
+
+
+@pytest.fixture
+def adobe_text_styling_postprocessor() -> AdobeTextStylingPostProcessor:
+    return AdobeTextStylingPostProcessor()
+
+
+@pytest.fixture
+def adobe_hyphenation_postprocessor() -> HyphenationPostProcessor:
+    return HyphenationPostProcessor()
+
+
+def test_hyphenation_postprocessor_rewrap_hyphenated_words(adobe_hyphenation_postprocessor):
```

Comment:
I deliberately used a case here that has some arbitrary problems (truncating at hyphens for some blocks, not others). The functionality I'm testing still passes, but I thought it would be useful to show just how arbitrary the Adobe API's outputs can be. This can also be seen in some other test cases. See comments for more detail.

Reply:
👍🏼